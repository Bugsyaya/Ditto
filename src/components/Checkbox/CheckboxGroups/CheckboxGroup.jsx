import React from 'react';
import { Checkbox as AntdCheckbox } from 'antd'

const CheckboxGroup = ({
  props,
  defaultValue = [],
  disabled = false,
  name,
  options = [],
  value = [],
  onChange,
}) => {
  return <AntdCheckbox.Group
    role='checkbox-group'
    defaultValue={defaultValue}
    name={name}
    disabled={disabled}
    options={options}
    value={value}
    onChange={onChange}
    {...props} />
}

export { CheckboxGroup };