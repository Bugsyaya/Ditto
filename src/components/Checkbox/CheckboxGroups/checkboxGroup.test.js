import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';
import React from 'react';
import {CheckboxGroup} from './CheckboxGroup';

describe('CheckboxGroup component', () => {
  const checkboxGroup = (
    <CheckboxGroup />
  );

  it('Snapshot CheckboxGroup', () => {
    const { baseElement } = render(checkboxGroup);
    expect(baseElement).toMatchSnapshot();
  });

  it('should render given children', () => {
    render(checkboxGroup);
    expect(screen.getByRole('checkbox-group')).toBeInTheDocument();
  });
});