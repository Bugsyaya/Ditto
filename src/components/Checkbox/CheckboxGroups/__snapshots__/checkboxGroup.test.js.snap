// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroup component Snapshot CheckboxGroup 1`] = `
<body>
  <div>
    <div
      class="ant-checkbox-group"
      role="checkbox-group"
    />
  </div>
</body>
`;
